### PR TITLE
Feat/vite react migration

### PR DESCRIPTION
**Pull request by Dsalazar1685**
-     Rebuilds CODA using Vite for easier bundling and faster hot reloading.
-     Upgrades React version and refactors existing components to use hooks instead of class components.
-     Upgrades React Router and replaces in-component router references with navigation hooks
-     Upgrades additional dependencies to latest versions.

### CODE REVIEW

File: .gitignore copy
Comment (fswiecki) [inline .gitignore copy:1]: This should probably be in the normal .gitignore?

File: package.json [hunk 1]
```diff
@@ -1,75 +1,44 @@
 {
-  "name": "coda",
-  "version": "1.0.0",
-  "description": "California Oak Disease and Arthropod Database",
-  "main": "/src/index.jsx",
+  "name": "coda-client-v2-temp",
```
Comment (fswiecki) [inline package.json:2]: could probably restore the head matter (coda, 1.0.0, description field, etc)

File: public/vite.svg
Comment (fswiecki) [inline public/vite.svg:1]: do we need the vite and react svgs?

**Review by fswiecki — APPROVED** [review]

File: .gitignore copy
Comment (Dsalazar1685) [reply .gitignore copy:1]: Good point, moved to normal .gitignore

File: public/vite.svg
Comment (Dsalazar1685) [reply public/vite.svg:1]: Nope, deleted now

File: package.json (hunk 1 above)
Comment (Dsalazar1685) [reply package.json]: Done

**Pull request merged**